validator/client/src/main/java/tech/pegasys/teku/validator/client/AbstractDutyScheduler.java

```diff
@@ -174,15 +174,26 @@ protected void onProductionDue(final UInt64 slot) {
       return;
     }
 
-    if (!isAbleToVerifyEpoch(slot)) {
-      LOG.info(
-          "Not performing {} duties for slot {} because it is too far ahead of the current slot {}",
-          dutyType,
+    if (lastProductionSlot != null && slot.compareTo(lastProductionSlot) == 0) {
+      LOG.debug(
+          "{} duties for slot {} already performed, last production slot {}",
+          dutyType.substring(0, 1).toUpperCase() + dutyType.substring(1),
           slot,
-          getCurrentEpoch().map(UInt64::toString).orElse("UNDEFINED"));
+          lastProductionSlot);
+      return;
+    }
```

Comment:
I worry about making this code more complex just to support better debug log messages.  Getting it wrong would create a slashable condition (slashing protection would stop it but we shouldn't need to depend on that in normal operation).  The code as written is correct but I'd prefer to keep it as simple as possible.

And actually in both cases the assumption is that the duties have already been performed because they would normally be done in sequence.